### DISCUSSION TITLE
How to stop mason to install packages

### DISCUSSION THREAD

**Discussion by BtAbdul** [LazyVim/LazyVim#1954] [Q&A]
**how i can stop mason to install these Packages?**

![IMG_20231105_210021](https://github.com/LazyVim/LazyVim/assets/46757797/bbefbca6-5d9c-48b8-a219-925315b617ad)

It automatically starts installing when i open LazyVim.
These packages is not supported in android,

I don't want to install them, i tried to delete theme from ensure_install mason but i didn't see any that in config.
or i try to add in skipped_install but i couldn't get any file where i can add them,

Please help me

**Comment by dpetka2001** [comment]
Those get installed from the `LazyExtras` that you have enabled in your other post. Please disable all extras, so that the packages don't get installed by mason.

**Reply by BtAbdul** [reply]
how i can modify mason plugin. To add theme in skipped servers ?  I try to add mason.lua file in config and inside i add these server in skipped server but it doesn't stop to download

**Reply by dpetka2001** [reply]
Did you disable the Extras? If you disable them, they won't get installed.

**Reply by BtAbdul** [reply]
i disabled them but some language i want to keep for example python, i want to keep python lsp, linters, formatting dap etc, i use alternative packages like pyright, flake isort etc,

Make them disable them python stuff gone

**Reply by BtAbdul** [reply]
If ruff is not supported its OK i will use another alternative package

**Reply by dpetka2001** [reply]
Oh I see, you could try something like this for the LSP servers to not be installed by mason
```lua
return {
  {
    "neovim/nvim-lspconfig",
    opts = {
      servers = {
        lua_ls = {
          mason = false,
        },
      },
    },
  },
}
```
You would have to this for every LSP server you want to not be installed by mason. Put this in a file in `~/.config/nvim/lua/plugins/lsp.lua`

**Reply by BtAbdul** [reply]
Thank you so much 😊

**Reply by keoren3** [reply]
@dpetka2001 Is there a chance this could be added in the formal manual?
It was quite a pain to look for it.

**Reply by folke** [reply]
It's already in the docs under lsp config.

**Reply by dpetka2001** [reply]
@keoren3 Like the maintainer already mentioned it's already in the [docs](https://www.lazyvim.org/plugins/lsp). If you see the configuration of `lua_ls` it mentions

> servers = {
      lua_ls = {
        -- mason = false, -- set to false if you don't want this server to be installed with mason

**Reply by devhaocui** [reply]
> Oh I see, you could try something like this for the LSP servers to not be installed by mason
> 
> ```lua
> return {
>   {
>     "neovim/nvim-lspconfig",
>     opts = {
>       servers = {
>         lua_ls = {
>           mason = false,
>         },
>       },
>     },
>   },
> }
> ```
> 
> You would have to this for every LSP server you want to not be installed by mason. Put this in a file in `~/.config/nvim/lua/plugins/lsp.lua`

OH MY GOD THANK YOU THANK YOU THANK YOUUUU SO MUCH YOU WOULD NOT HAVE BELIEVED HOW MANY HOURS I LOOKED FOR THIS. Gosh dang lazy.nvim keeps reinstalling it with mason no matter what i did, I tried doing configs with mason-lspconfig and tried to mess with mason's configs then tried a boat load of stuff. Thank you!!!!!!

**Reply by Flimm** [reply]
This worked well for me. Don't forget to uninstall Mason's previously installed ruff by opening Neovim, running `:Mason`, selecting `ruff`, and pressing `X`, or by running `:MasonUninstallAll`

**Comment by binhtddev** [comment]
I'm using Nix and here is my config to control which I really want to install packages in Mason. https://github.com/binhtran432k/dotfiles/blob/a264fce08dd931e40bf47bd364dc80a8e55cc1bb/nvim/lua/plugins/lang.lua#L46-L68
Note: I have a list of packages which is not available in nix, please remove them to avoid install in Mason.

**Comment by nimeshmaharjan1** [comment]
how do you even uninstall the packages from mason?

- this prettierd is formatting completely differently than the vscode and even have the .pretterc file
- how do i uninstall it the prettierrd now?

![image](https://github.com/user-attachments/assets/f664217c-a6d5-455d-aae0-0c8b179a9927)

**Reply by devhaocui** [reply]
for that specific photo you share you can scroll up or down to the package you want to uninstall then press x or X. for installing packages goto the ones that aren't installed and press i

**Reply by dpetka2001** [reply]
You can just press `g?` like it's mentioned in the Mason UI to switch to the help page where it also mentions the mappings.

**Reply by nimeshmaharjan1** [reply]
> for that specific photo you share you can scroll up or down to the package you want to uninstall then press x or X. for installing packages goto the ones that aren't installed and press i 

thank you i knew it was i for installing but could not find for uninstalling it

**Reply by nimeshmaharjan1** [reply]
> You can just press `g?` like it's mentioned in the Mason UI to switch to the help page where it also mentions the mappings.

x worked for me or i simply did MasonUnistall but i was looking at the mappings in the lazyvim website
